netmaster/master/network.go

```diff
@@ -163,6 +164,57 @@ func deleteDockNet(networkName string) error {
 	return nil
 }
 
+func detachServiceContainer(networkName, tenantName string) error {
+	docker, err := utils.GetDockerClient()
+	if err != nil {
+		log.Errorf("Unable to connect to docker. Error %v", err)
+		return errors.New("Unable to connect to docker")
+	}
+
+	dnsContName := tenantName + "dns"
+	cinfo, err := docker.InspectContainer(dnsContName)
+	if err != nil {
+		log.Errorf("Error inspecting the container %s. Err: %v", dnsContName, err)
+		return err
+	}
+
+	nwState, err := docker.InspectNetwork(networkName)
+	if err != nil {
+		log.Errorf("Error while inspecting network: %+v", networkName)
+		return err
+	}
+	log.Infof("Containers in network: %+v are {%+v}", networkName, nwState.Containers)
+	dnsServerIP := strings.Split(nwState.Containers[cinfo.Id].IPv4Address, "/")[0]
+
+	stateDriver, err := utils.GetStateDriver()
+	if err != nil {
+		log.Errorf("Could not get StateDriver while trying to disconnect dnsContainer from %+v", networkName)
+		return err
+	}
+
+	// Read network config and get DNSServer information
+	nwCfg := &mastercfg.CfgNetworkState{}
+	nwCfg.StateDriver = stateDriver
+	err = nwCfg.Read(networkName)
+	if err != nil {
+		return err
+	}
+
+	log.Infof("dnsServerIP: %+v, nwCfg.dnsip: %+v", dnsServerIP, nwCfg.DNSServer)
+	// Remove dns container from network if all other endpoints are withdrawn
+	if len(nwState.Containers) == 1 && (dnsServerIP == nwCfg.DNSServer) {
```

Comment:
I'm working on more changes to handle attaching/detaching service containers in a more elegant way. This will probably change as part of that. Will make the change accordingly in the next PR.